lib/active_admin/views/components/index_list.rb

```diff
@@ -36,7 +36,9 @@ def build(index_classes)
       # @param [Class] index_class The class on which to build the link and html classes
       def build_index_list(index_class)
         li class: classes_for_index(index_class) do
-          a href: url_for(as: index_class.index_name.to_sym), class: "table_tools_button" do
+          params = request.query_parameters.except :page, :commit, :format
+
+          a href: url_for(params.merge(as: index_class.index_name.to_sym)), class: "table_tools_button" do
```

Comment:
please move `params.merge(as: index_class.index_name.to_sym)` to a separate line.